### PR TITLE
m46c verbs -> hud buttons

### PR DESCRIPTION
**Pull request by stanalbatross**
# About the pull request

converted the m46c's verbs (toggle id lock and toggle iff) to hud buttons

image

like so

also added toggle iff hud button to the CO SG

and removed a stray nanoui file I missed earlier

# Explain why it's good for the game

hud buttons are significantly easier to use than verbs + fuck verbs

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
qol: converted the M46c's verbs to hud buttons. Sprites by Nauticall.
qol: added a "toggle id lock" hud button to the CO smartgun
/:cl:

### CODE REVIEW

**Review by morrowwolf — CHANGES_REQUESTED** [review]
Make a hotkey for IFF toggle.

File: code/modules/projectiles/guns/rifles.dm [hunk 1]
```diff
@@ -473,35 +479,82 @@
...
+	action_icon_state = "iff_toggle_on"
+	button.name = name
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/hud/actions.dmi', button, action_icon_state)
 
+/datum/action/item_action/m46c/toggle_lethal_mode/action_activate()
+	. = ..()
+	var/obj/item/weapon/gun/rifle/m46c/G = holder_item
+	G.toggle_iff(usr)
+	if(G.iff_enabled)
```
Comment (harryob) [inline code/modules/projectiles/guns/rifles.dm:511]: G 💢💢💢

File: code/modules/projectiles/guns/rifles.dm [hunk 2]
```diff
@@ -473,35 +479,82 @@
...
+	action_icon_state = "id_lock_locked"
+	button.name = name
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/hud/actions.dmi', button, action_icon_state)
+
+/datum/action/item_action/m46c/toggle_id_lock/action_activate()
+	. = ..()
+	var/obj/item/weapon/gun/rifle/m46c/G = holder_item
+	G.toggle_lock()
+	if(G.is_locked)
```
Comment (harryob) [inline code/modules/projectiles/guns/rifles.dm:530]: 💢💢💢

File: code/modules/projectiles/guns/smartgun.dm [hunk 3]
```diff
@@ -551,37 +551,80 @@
...
-		to_chat(usr, SPAN_NOTICE("[icon2html(src)] You pick up \the [src], registering yourself as its owner."))
-	..()
+// ID lock action \\
+
+/datum/action/item_action/co_sg/action_activate()
+	var/obj/item/weapon/gun/smartgun/co/G = holder_item
+	if(!ishuman(owner))
+		return
+	var/mob/living/carbon/human/H = owner
+	if(H.is_mob_incapacitated() || G.get_active_firearm(H, FALSE) != holder_item)
```
Comment (harryob) [inline code/modules/projectiles/guns/smartgun.dm:583]: protagonist_gun*, protagonist*

File: code/modules/projectiles/guns/smartgun.dm [hunk 4]
```diff
@@ -551,37 +551,80 @@
...
+	action_icon_state = "id_lock_locked"
+	button.name = name
+	button.overlays.Cut()
+	button.overlays += image('icons/mob/hud/actions.dmi', button, action_icon_state)
+
+/datum/action/item_action/co_sg/toggle_id_lock/action_activate()
+	. = ..()
+	var/obj/item/weapon/gun/smartgun/co/G = holder_item
+	G.toggle_lock()
+	if(G.is_locked)
```
Comment (harryob) [inline code/modules/projectiles/guns/smartgun.dm:601]: as above

**Review by harryob — CHANGES_REQUESTED** [review]

**Comment by harryob** [comment]
imagine forgetting

**Review by harryob — APPROVED** [review]

File: code/modules/projectiles/guns/smartgun.dm [hunk 5]
```diff
@@ -551,37 +551,80 @@
 	var/mob/living/carbon/human/linked_human
 	var/is_locked = TRUE
 
+	actions_types = list(
```
Comment (morrowwolf) [inline code/modules/projectiles/guns/smartgun.dm:554]: List should be defined in Initialization.

File: code/modules/projectiles/guns/smartgun.dm [hunk 6]
```diff
@@ -551,37 +551,80 @@
...
+/obj/item/weapon/gun/smartgun/co/pickup(user)
+	if(!linked_human)
+		src.name_after_co(user, src)
+		to_chat(usr, SPAN_NOTICE("[icon2html(src, usr)] You pick up \the [src], registering yourself as its owner."))
+	..()
+
+
 /obj/item/weapon/gun/smartgun/co/proc/name_after_co(var/mob/living/carbon/human/H, var/obj/item/weapon/gun/smartgun/co/I)
 	linked_human = H
 	RegisterSignal(linked_human, COMSIG_PARENT_QDELETING, .proc/remove_idlock)
```
Comment (morrowwolf) [inline code/modules/projectiles/guns/smartgun.dm:630]: Github won't let me go down any further but get_examine_text() (line 632) needs to be converted to the new system as it is only sending the to_chat() about registered user. See M46C for example.

**Review by morrowwolf — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/rifles.dm [hunk 7]
```diff
@@ -424,8 +424,14 @@
 	auto_retrieval_slot = WEAR_J_STORE
 	map_specific_decoration = TRUE
 
+	actions_types = list(
```
Comment (morrowwolf) [inline code/modules/projectiles/guns/rifles.dm:427]: List declaration in initialization

**Review by morrowwolf — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/smartgun.dm [hunk 8]
```diff
@@ -551,17 +551,8 @@
...
-						/datum/action/item_action/smartgun/toggle_ammo_type,
-						/datum/action/item_action/smartgun/toggle_auto_fire,
-						/datum/action/item_action/smartgun/toggle_lethal_mode,
-						/datum/action/item_action/smartgun/toggle_motion_detector,
-						/datum/action/item_action/smartgun/toggle_recoil_compensation,
-						/datum/action/item_action/co_sg/toggle_id_lock
-						)
-
 /obj/item/weapon/gun/smartgun/co/able_to_fire(mob/user)
+	LAZYADD(actions_types, /datum/action/item_action/co_sg/toggle_id_lock)
```
Comment (morrowwolf) [inline code/modules/projectiles/guns/smartgun.dm:555]: This is not initialize 👀

**Review by morrowwolf — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/guns/smartgun.dm (hunk 8 above)
Comment (stanalbatross) [reply code/modules/projectiles/guns/smartgun.dm:555]: my brain was on autopilot

**Review by morrowwolf — APPROVED** [review]

**Pull request merged**